scripts/admin/launch-meteor

```diff
@@ -82,15 +82,36 @@ if [ ! -x "$METEOR_WAREHOUSE_DIR/meteor" ]; then
   fi
   echo "Installing a Meteor distribution in your home directory." 1>&2
 
-  # Only show progress bar animations if we have a tty
-  # (Prevents tons of console junk when installing within a pipe)
-  if [[ -t 1 ]]; then
-    curl --progress-bar --fail "$TARBALL_URL" | tar -xzf - -C "$INSTALL_TMPDIR"
-  else
-    curl -s --fail "$TARBALL_URL" | tar -xzf - -C "$INSTALL_TMPDIR"
-  fi
+  TARBALL_FILE="$(dirname "$METEOR_WAREHOUSE_DIR")/.meteor-tarball-tmp"
+  rm -rf "$TARBALL_FILE"
+
+  echo "Downloading Meteor distribution"
+  # keep trying to curl the file until it works (resuming where possible)
+  MAX_ATTEMPTS=99
+  set +e
+  CODE=1
+  ATTEMPTS=0
+  while [ $CODE -ne 0 ] && [ $ATTEMPTS -lt $MAX_ATTEMPTS ]
+  do
+    ATTEMPTS=$((ATTEMPTS + 1))
+
+    # Only show progress bar animations if we have a tty
+    # (Prevents tons of console junk when installing within a pipe)
+    if [[ -t 1 ]]; then
```

Comment:
The rest of the additions look POSIX-compatible, but because `/bin/sh` isn't always `bash` (GNU), but sometimes `dash` [such as on Ubuntu](https://wiki.ubuntu.com/DashAsBinSh), the `[[` will break things (and currently is, see #7353).